samcli/lib/providers/sam_api_provider.py

```diff
@@ -98,6 +117,179 @@ def _extract_from_serverless_api(
         collector.stage_variables = stage_variables
         collector.cors = cors
 
+        auth = properties.get(SamApiProvider._AUTH, {})
+        if not auth:
+            return
+
+        default_authorizer = auth.get(SamApiProvider._DEFAULT_AUTHORIZER)
+        if default_authorizer:
+            collector.set_default_authorizer(logical_id, default_authorizer)
+
+        self._extract_authorizers_from_props(logical_id, auth, collector, Route.API)
+
+    @staticmethod
+    def _extract_request_lambda_authorizer(
+        auth_name: str, function_name: str, prefix: str, properties: dict, event_type: str
+    ) -> LambdaAuthorizer:
+        """
+        Generates a request Lambda Authorizer from the given identity object
+
+        Parameters
+        ----------
+        auth_name: str
+            Name of the authorizer
+        function_name: str
+            Name of the Lambda function this authorizer uses
+        prefix: str
+            The prefix to prepend to identity sources
+        properties: dict
+            The authorizer properties that contains identity sources and authorizer specific properties
+        event_type: str
+            The type of API this is (API or HTTP API)
+
+        Returns
+        -------
+        LambdaAuthorizer
+            The request based Lambda Authorizer object
+        """
+        payload_version = properties.get(SamApiProvider._AUTHORIZER_PAYLOAD)
+
+        if payload_version is not None and not isinstance(payload_version, str):
```

Comment:
I wonder for validation overall, for this PR and other PRs, it might be easier if we have a validation decorator with a certain set of rules that we can attach to a given function. This gives a cleaner delineation between validation logic and the business logic.

Looks to me, most of the validation is around the args themselves.

Reply:
Thats a good callout about validation vs business logic. Its not ideal since we in each instance (authorizers under `Properties`, as a CFN resource, under Swagger) very slight variations in the keys and some slight variation in the expected type/format.

Moving this validation logic out is probably best done once that other PR is merged in to reuse that class.